### PR TITLE
games.zetsubou.SablesGrimoireDemo

### PR DESCRIPTION
**Pull request by koroshiya**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

Additional notes:

- this game contains a minimal set of python libraries needed to run. It doesn't require python2.7 itself to be included in the manifest, however, as the game engine has its own executable which can run python code.
- I confirm that I have permission to distribute this game. I haven't made the game open source, but I am the author of this game, and I intend for this to be an official distribution.
- I confirm that I am submitting the desktop file, icon and metainfo files upstream

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build games.zetsubou.SablesGrimoireDemo

File: games.zetsubou.SablesGrimoireDemo.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+app-id: games.zetsubou.SablesGrimoireDemo
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
```
Comment (hfiguiere) [inline games.zetsubou.SablesGrimoireDemo.yml:3]:
```suggestion
runtime-version: '22.08'
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]
have the desktop icon and appstream file been upstreamed?

File: games.zetsubou.SablesGrimoireDemo.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          - aarch64
+      - type: file
+        path: games.zetsubou.SablesGrimoireDemo.desktop
+        sha512: 585bb7985af9884d87a33d3b3f3b53b70103f267d5c049038fa07665c07f5a782fbc1e31f9795195814f28242363095e771657151f967ecb9ff4459b06001fb4
+      - type: file
+        path: games.zetsubou.SablesGrimoireDemo.png
+        sha512: 90e252aa66d89e9fa1a03333d40465273209ff505d3a006346f8b20337f101eae3896663abc1344cf003ebe78fe85bd5edba1050991e225e58da7fa0cb339e1b
+      - type: file
+        path: games.zetsubou.SablesGrimoireDemo.metainfo.xml
+        sha512: 1f92ee48029d8e06b7a32311013352bd8310d7ef8f3a784eb0266b8d5db0668a05328109d21d00ed7625f7c3614c2ef4da19fa77c3f564a1d72998f504b5ca55
```
Comment (hfiguiere) [inline games.zetsubou.SablesGrimoireDemo.yml:42]:
```suggestion
```

File: games.zetsubou.SablesGrimoireDemo.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        url: https://cdn.zetsubou.games/Flatpak/SablesGrimoireDemo/builds/v1.2.0-aarch64.tar.bz2
+        sha512: 7eedd2b06234f919a939dbfb96ac9162efa7b220592bd6fcf4f37eaa52db57c444a456372be586f88bc07a4f2039e70cfd8bff1487f32d40c627cf09937ca902
+        only-arches:
+          - aarch64
+      - type: file
+        path: games.zetsubou.SablesGrimoireDemo.desktop
+        sha512: 585bb7985af9884d87a33d3b3f3b53b70103f267d5c049038fa07665c07f5a782fbc1e31f9795195814f28242363095e771657151f967ecb9ff4459b06001fb4
+      - type: file
+        path: games.zetsubou.SablesGrimoireDemo.png
+        sha512: 90e252aa66d89e9fa1a03333d40465273209ff505d3a006346f8b20337f101eae3896663abc1344cf003ebe78fe85bd5edba1050991e225e58da7fa0cb339e1b
```
Comment (hfiguiere) [inline games.zetsubou.SablesGrimoireDemo.yml:39]:
```suggestion
```

File: games.zetsubou.SablesGrimoireDemo.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        only-arches:
+          - x86_64
+      - type: archive
+        url: https://cdn.zetsubou.games/Flatpak/SablesGrimoireDemo/builds/v1.2.0-aarch64.tar.bz2
+        sha512: 7eedd2b06234f919a939dbfb96ac9162efa7b220592bd6fcf4f37eaa52db57c444a456372be586f88bc07a4f2039e70cfd8bff1487f32d40c627cf09937ca902
+        only-arches:
+          - aarch64
+      - type: file
+        path: games.zetsubou.SablesGrimoireDemo.desktop
+        sha512: 585bb7985af9884d87a33d3b3f3b53b70103f267d5c049038fa07665c07f5a782fbc1e31f9795195814f28242363095e771657151f967ecb9ff4459b06001fb4
```
Comment (hfiguiere) [inline games.zetsubou.SablesGrimoireDemo.yml:36]:
```suggestion
```

**Comment by koroshiya** [comment]
> have the desktop icon and appstream file been upstreamed?

Yes

**Comment by hfiguiere** [comment]
bot, build games.zetsubou.SablesGrimoireDemo

**Comment by hfiguiere** [comment]
when I try to run this I get:

```
[S_API] SteamAPI_Init(): SteamAPI_IsSteamRunning() did not locate a running instance of Steam.
dlopen failed trying to load:
/home/hub/.steam/sdk64/steamclient.so
with error:
/home/hub/.steam/sdk64/steamclient.so: cannot open shared object file: No such file or directory
[S_API] SteamAPI_Init(): Sys_LoadModule failed to load: /home/hub/.steam/sdk64/steamclient.so
```

I starts though.

**Comment by koroshiya** [comment]
> when I try to run this I get:
> 
> ```
> [S_API] SteamAPI_Init(): SteamAPI_IsSteamRunning() did not locate a running instance of Steam.
> dlopen failed trying to load:
> /home/hub/.steam/sdk64/steamclient.so
> with error:
> /home/hub/.steam/sdk64/steamclient.so: cannot open shared object file: No such file or directory
> [S_API] SteamAPI_Init(): Sys_LoadModule failed to load: /home/hub/.steam/sdk64/steamclient.so
> ```
> 
> I starts though.

This is expected and shouldn't break anything.
If steam is running and detected, and this game is in their steam library, then it should integrate with certain steam features.
I don't know if the steam integration actually works within a flatpak though.

**Pull request closed without merging**